tests/publishTests.js

```diff
@@ -0,0 +1,350 @@
+});
+
+// publish.getDevVersion
+console.log("\n*** publish.getDevVersion ***");
+
+var getDevVersionFixture = [{
+    rawTimestamp: "get raw timestamp",
+    revision: "get revision",
+    devVersion: "${timestamp}.${revision}",
+    expectedVersion: "20151015T131223Z.039d221",
+    returnedTimestamp: 1444914743,
+    returnedRevision: "039d221"
+}];
+
+getDevVersionFixture.forEach(function (fixture) {
+    console.log("getDevVersion test - rawTimestamp: " + fixture.rawTimestamp + " revision: " + fixture.revision + " devVersion: " + fixture.devVersion);
+
+    var exec = sinon.stub(publish, "execSync");
+    exec.onFirstCall().returns(fixture.returnedTimestamp);
+    exec.onSecondCall().returns(fixture.returnedRevision);
+
+    var result = publish.getDevVersion(fixture);
+
+    assert(exec.calledTwice, "execSync should have been called twice");
+    assert(exec.calledWith(fixture.rawTimestamp), "first execSync should have been called with: " + fixture.rawTimestamp);
+    assert(exec.calledWith(fixture.revision), "second execSync should have been called with: " + fixture.revision);
+    assert.equal(result, fixture.expectedVersion, "Expected version: " + fixture.expectedVersion + " actual: " + result);
+
+    // remove execSync stub
+    publish.execSync.restore();
+});
+
+// publish.pubImpl
+console.log("\n*** publish.pubImpl ***");
+
+var pubImplFixture = [{
+    isTest: true,
+    pack: "pack",
+    publish: "shouldn't publish"
+}, {
+    isTest: false,
+    pack: "shouldn't pack",
+    publish: "publish"
+}, {
+    pack: "shouldn't pack",
+    publish: "publish"
+}];
+
+pubImplFixture.forEach(function (fixture) {
+    console.log("pubImpl test - isTest: " + fixture.isTest + " pack: " + fixture.pack + " publish: " + fixture.publish);
+
+    var exec = sinon.stub(publish, "execSync");
+    var expected = fixture[fixture.isTest ? "pack" : "publish"];
+
+    publish.pubImpl(fixture.isTest, fixture);
+    assert(exec.calledOnce, "execSync should have been called");
+    assert(exec.calledWith(expected), "execSync should have been called with: " + expected);
+
+    // remove execSync stub
+    publish.execSync.restore();
+});
+
+// publish.tag
+console.log("\n*** publish.tag ***");
+
+var tagFixture = [{
+    isTest: true,
+    version: "1.0.0",
+    tag: "tag",
+    distTag: "add tag ${tag} to ${version}",
+    expected: "tag command: add tag tag to 1.0.0"
+}, {
+    isTest: false,
+    version: "2.0.0",
+    tag: "tag2",
+    distTag: "add tag ${tag} to ${version}",
+    expected: "add tag tag2 to 2.0.0"
+}, {
+    version: "3.0.0",
+    tag: "tag3",
+    distTag: "add tag ${tag} to ${version}",
+    expected: "add tag tag3 to 3.0.0"
+}];
+
+tagFixture.forEach(function (fixture) {
+    console.log("tag test - isTest: " + fixture.isTest + " version: " + fixture.version + " tag: " + fixture.tag + " distTag: " + fixture.distTag);
+
+    var exec = sinon.stub(publish, "execSync");
+    var log = sinon.stub(console, "log");
```

Comment:
This one is a little disturbing and will punish your own ability to call console.log from test fixtures. Given your "stubbability" of publish.execSync worked so will, I suggest you follow the same approach there and hoist up the module's console.log to publish.log and stub that one instead, rather than polluting the global console.log